### PR TITLE
Return empty .json() object on 204. Fix #165.

### LINKED ISSUE #165: Return empty object on 204 No Content...
...instead of throwing a syntax error due to attempting to JSON parse `''`.

```
let fetch = require('node-fetch');
fetch('http://httpbin.org/status/204').then(res => res.json()).then(object => {
    console.log(object);
}).catch(error => console.error(error));
```

Demo: http://runkit.com/dandv/node-fetch-204-no-content

Fixing this would attain compatibility with [request](https://github.com/request/request/issues/2177) and [got](https://github.com/sindresorhus/got/pull/78).

More generally, it would be more tolerant to return an empty object if the response is the empty string (regardless of the status code being 204 or other 2xx).

### PR DESCRIPTION
**Pull request by dandv**
+ some cosmetic whitespace fixes

### CODE REVIEW

**Comment by bitinn** [comment]
I agree 204 empty response should be handled, but I think checking `this.status == 204` is a better solution.

I don't think it's `node-fetch`'s responsibility to handle json() on empty string. Or at least Fetch Spec doesn't seem to give any special treatment for it.

https://fetch.spec.whatwg.org/#dom-body-bodyused

**Comment by dandv** [comment]
> I think checking this.status == 204 is a better solution

I thought so too, but opted to check for the actual empty string, regardless of the response code, because that's more resilient and the actual source of the problem. Is there a use case for failing with a syntax error on a 200 response that's empty because the implementor of the service didn't return 204?

**Comment by bitinn** [comment]
> because that's more resilient and the actual source of the problem

I agree to some degree, for maximum compatibility, just checking empty string makes it work better across different scenarios.

But as mentioned in `got`, returning a 200 empty response when users are requesting JSON, is simply invalid, and we are right to throw an error.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "node-fetch",
-  "version": "1.6.1",
+  "version": "1.6.2",
```
Comment (bitinn) [inline package.json]: Could you kindly revert this change, we actually bump version using `npm version` right before publish.

File: lib/body.js [hunk 2]
```diff
@@ -39,7 +39,9 @@ function Body(body, opts) {
  */
 Body.prototype.json = function() {
 
-	return this._decode().then(function(buffer) {
+	
+	if (this.status === 204) return Body.Promise.resolve({});
+        return this._decode().then(function(buffer) {
```
Comment (bitinn) [inline lib/body.js]: Please format this line as tab.

File: test/test.js [hunk 3]
```diff
@@ -1402,7 +1413,8 @@ describe('node-fetch', function() {
 		return req.buffer().then(function(result) {
 			expect(result.toString()).to.equal('a=1');
 		});
-	}); 
+	});
+
```
Comment (bitinn) [inline test/test.js]: And I have no idea why these changes are needed?

Comment (dandv) [inline test/test.js]: Those were trailing whitespace automatically removed by my editor.

Comment (bitinn) [inline test/test.js]: but it also adds an additional newline.

File: package.json (hunk 1 above)
Comment (dandv) [inline package.json]: Fixed. I thought GitHub would hide the discussion now due to it being on an outdated diff, but I still see it.

File: test/test.js (hunk 3 above)
Comment (dandv) [inline test/test.js]: I might have to file a bug with the Kate editor about that... :-/

File: lib/body.js (hunk 2 above)
Comment (dandv) [inline lib/body.js]: Done.

**Comment by bitinn** [comment]
released in 1.6.2
